services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -0,0 +1,102 @@
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.service.lockkeeper.LockKeeperService
+import org.openkilda.testing.service.northbound.NorthboundService
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+
+class SwitchRulesSpec extends BaseSpecification {
+
+    @Autowired
+    TopologyDefinition topology
+    @Autowired
+    NorthboundService northboundService
+    @Autowired
+    LockKeeperService lockKeeperService
+
+    @Value('${reroute.delay}')
+    int rerouteDelay
+    @Value('${discovery.timeout}')
+    int discoveryTimeout
+
+    def "Default rules are installed on a new switch when connecting it to the controller"() {
+        requireProfiles("virtual")
+
+        given: "A switch with no rules installed and not connected to the controller"
+        def sw = topology.getActiveSwitches().first()
+        def defaultRulesSorted = northboundService.getSwitchRules(sw.dpId).flowEntries.sort { it.cookie }
+        northboundService.deleteSwitchRules(sw.dpId, DeleteRulesAction.DROP_ALL)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getSwitchRules(sw.dpId).flowEntries.isEmpty() }
+
+        lockKeeperService.knockoutSwitch(sw.dpId)
+        assert Wrappers.wait(WAIT_OFFSET) { !(sw.dpId in northboundService.getActiveSwitches()*.switchId) }
+
+        when: "Connect the switch to the controller"
+        lockKeeperService.reviveSwitch(sw.dpId)
+
+        then: "The switch is really connected to the controller"
+        Wrappers.wait(WAIT_OFFSET) { sw.dpId in northboundService.getActiveSwitches()*.switchId }
+
+        and: "Default rules are installed on the switch"
+        def actualRulesSorted = northboundService.getSwitchRules(sw.dpId).flowEntries.sort { it.cookie }
+        actualRulesSorted.size() == defaultRulesSorted.size()
+        [0..actualRulesSorted.size() - 1].each { i ->
+            with(actualRulesSorted[i]) {
+                verifyAll {
+                    cookie == defaultRulesSorted[i].cookie
+                    tableId == defaultRulesSorted[i].tableId
+                    version == defaultRulesSorted[i].version
+                    priority == defaultRulesSorted[i].priority
+                    idleTimeout == defaultRulesSorted[i].idleTimeout
+                    hardTimeout == defaultRulesSorted[i].hardTimeout
+                    match == defaultRulesSorted[i].match
+                    instructions == defaultRulesSorted[i].instructions
+                    flags == defaultRulesSorted[i].flags
+                }
+            }
+        }
+    }
+
+    def "Pre-installed rules are not deleted from a new switch when connecting it to the controller"() {
+        requireProfiles("virtual")
+        assumeTrue("Test is skipped because of the issue #1464. Also the current disconnection of the switch from " +
```

Comment:
This is better to be described in `@Ignore("why?")`. Assumptions better fit for things that can only be computed at runtime